### PR TITLE
Add Email and Registry lookups to Address Selector

### LINKED ISSUE #3871: Address box should support email and registered names
Address box currently only supports hex addresses and local address book/account names. It should support registered names (made through our Registry contract) and emails (through our ProofOfEmail contract).

### PR DESCRIPTION
**Pull request by ngotchac**
Closes #3871 

Adds to the new Address Selector lookups capabilities from the Registry and the Email Verification contract.

The result is shown right under the input.

mail address selector

It also fixes some issues in the component.

### CODE REVIEW

File: js/src/ui/Form/AddressSelect/addressSelect.js [hunk 1]
```diff
@@ -225,10 +225,8 @@ class AddressSelect extends Component {
     }
 
     const accounts = regsitryValues
-      .map((regsitryValue) => {
-        const { address, value } = regsitryValue;
-        const account = { address, name: value, index: address };
-
+      .map((regsitryValue, index) => {
```
Comment (derhuerst) [inline js/src/ui/Form/AddressSelect/addressSelect.js]: typo. 😉

File: js/src/ui/Form/AddressSelect/addressSelectStore.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    registry
+      .getContract('emailverification')
+      .then((emailVerification) => {
+        this.regLookups.push({
+          lookup: (value) => {
+            return emailVerification
+              .instance
+              .reverse.call({}, [ sha3(value) ]);
+          },
+          describe: (value) => `${value} (from email verification)`
```
Comment (derhuerst) [inline js/src/ui/Form/AddressSelect/addressSelectStore.js]: I'd prefer "Verified using email …", as this is being shown inside and account card.

File: js/src/ui/AccountCard/accountCard.css [hunk 3]
```diff
@@ -52,6 +52,11 @@
   }
 }
 
+.description {
+  font-size: 0.75em;
+  color: rgba(255, 255, 255, 0.498039);
```
Comment (jacogr) [inline js/src/ui/AccountCard/accountCard.css]: Make it 0.5 alpha, Chrome creates these numbers when inspecting.

File: js/src/ui/Form/AddressSelect/addressSelectStore.js (hunk 2 above)
Comment (jacogr) [inline js/src/ui/Form/AddressSelect/addressSelectStore.js]: While we are doing this fresh, create strings with `FormattedMessage` so we don't need to circle back. (Also for the labels in this store)

File: js/src/ui/Form/AddressSelect/addressSelectStore.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+    registry
+      .getInstance()
+      .then((registryInstance) => {
+        this.regLookups.push({
+          lookup: (value) => {
+            return registryInstance
+              .getAddress.call({}, [ sha3(value), 'A' ]);
+          },
+          describe: (value) => `${value} (from registry)`
```
Comment (jacogr) [inline js/src/ui/Form/AddressSelect/addressSelectStore.js]:
```
<FormattedMessage
  id='addressSelect.fromRegistry'
  defaultMessage='{value} (from registry)'
  values={ {
    value
  } } />
```

File: js/src/ui/Form/AddressSelect/addressSelectStore.js [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+  @action handleChange = (value = '') => {
+    let index = 0;
+
+    this.values = this.initValues
+      .map((category) => {
+        const filteredValues = this
+          .filterValues(category.values, value)
+          .map((value) => {
+            index++;
+            return { ...value, index: parseInt(index) };
```
Comment (jacogr) [inline js/src/ui/Form/AddressSelect/addressSelectStore.js]: Would prefer 

```
return {
  index: parseint(index),
  ...value
};
```

File: js/src/ui/Form/AddressSelect/addressSelectStore.js [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+      .all(lookups)
+      .then((results) => {
+        return results
+          .map((result, index) => {
+            if (/^(0x)?0*$/.test(result)) {
+              return;
+            }
+
+            const account = flatMap(this.initValues, (cat) => cat.values)
+              .find((account) => account.address.toLowerCase() === result.toLowerCase());
```
Comment (jacogr) [inline js/src/ui/Form/AddressSelect/addressSelectStore.js]: Do we need to lowercase result each callback? Could do it outside and assign to variable being tested.

File: js/src/ui/Form/AddressSelect/addressSelectStore.js [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+    const contactsN = Object.keys(contacts).length;
+    const walletsN = Object.keys(wallets).length;
+
+    if (accountsN + contractsN + contactsN + walletsN === 0) {
+      return;
+    }
+
+    this.initValues = [
+      {
+        label: 'accounts',
```
Comment (jacogr) [inline js/src/ui/Form/AddressSelect/addressSelectStore.js]: Labels as per comment above.

File: js/src/ui/Form/AddressSelect/addressSelect.js [hunk 8]
```diff
@@ -241,8 +217,28 @@ class AddressSelect extends Component {
...
   }
 
+  renderRegsitryValues () {
+    const { regsitryValues } = this.store;
+
+    if (regsitryValues.length === 0) {
+      return null;
+    }
+
+    const accounts = regsitryValues
```
Comment (jacogr) [inline js/src/ui/Form/AddressSelect/addressSelect.js]: `registryValues` (typo, also registryValue below as well as the function name)

**Comment by ngotchac** [comment]
Fixed

**Pull request merged**